api/v1alpha2/conversion_test.go

```diff
@@ -49,6 +51,44 @@ func TestConvertCluster(t *testing.T) {
 	})
 
 	t.Run("from hub", func(t *testing.T) {
+		t.Run("preserves the hub version", func(t *testing.T) {
+			src := &v1alpha3.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "hub",
+				},
+			}
+			dst := &Cluster{}
+
+			g.Expect(dst.ConvertFrom(src)).To(Succeed())
+			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("hub"))
+		})
+
+		t.Run("preserves Spec.ControlPlaneRef", func(t *testing.T) {
+			src := &v1alpha3.Cluster{
+				Spec: v1alpha3.ClusterSpec{
+					ControlPlaneRef: &corev1.ObjectReference{
+						Name: "controlplane-1",
+					},
+				},
+			}
+			dst := &Cluster{}
+
+			g.Expect(dst.ConvertFrom(src)).To(Succeed())
+			g.Expect(dst.GetAnnotations()[utilconversion.DataAnnotation]).To(ContainSubstring("controlplane-1"))
```

Comment:
I think I'd rather (or additionally) see a call to ConvertTo and then verify that the new v1a3 object has the expected fields set (could do this in a single test instead of one per field)